services/git-service-api/src/main/java/io/fabric8/launcher/service/git/api/GitRepositoryFilter.java

```diff
@@ -0,0 +1,19 @@
+package io.fabric8.launcher.service.git.api;
+
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable(singleton = true)
+@JsonSerialize(as = ImmutableGitRepositoryFilter.class)
+@JsonDeserialize(as = ImmutableGitRepositoryFilter.class)
+public interface GitRepositoryFilter {
+
+    @Nullable String withNameContaining();
+
+    @Value.Parameter
```

Comment:
Just curious, wouldn't adding a `@Parameter` here making it part of the `ImmutableGitRepositoryFilter.of` signature?

Reply:
yes (I wanted it as a shortcut since it's a common usage of this filter), and the singleton makes a of() (to build an empty filter). Do you see a reason why I shouldn't use it?

Reply:
You say it is a @Parameter but at the same time it is @Nullable, so it is confusing. @Parameters shouldn't be @Nullable afaik

Reply:
You say it is a @Parameter but at the same time it is @Nullable, so it is confusing. @Parameters shouldn't be @Nullable afaik

Reply:
I don't see any reason why a parameter shouldn't be @nullable, but let's use `Optional` if you would feel more comfy then lol

Reply:
I don't agree. According to best practices `Optional` should _not_ be used as parameters. And `@Nullable` parameters are perfectly normal

Reply:
@quintesse I may be wrong, but I don't think using Optional here will create it as a method parameter

Reply:
Using Optional and @parameter end up to this:
```Java
/**
   * Construct a new immutable {@code GitRepositoryFilter} instance.
   * @param withOrganization The value for the {@code withOrganization} attribute
   * @return An immutable GitRepositoryFilter instance
   */
  public static ImmutableGitRepositoryFilter of(Optional<GitOrganization> withOrganization) {
    return validate(new ImmutableGitRepositoryFilter(withOrganization));
  }

```

Reply:
Horrible. I thought that would work only for method returns